APIScrape/scrape.php

```diff
@@ -0,0 +1,153 @@
+    for ($i = 0; $i < count($responseArray); $i++) {
+        $associativeBreedArray[$responseArray[$i]["id"]] = $responseArray[$i]["name"];
+    }
+    return $associativeBreedArray;
+}
+
+/**
+ * Puts the list of breeds into the database.
+ * 
+ * @param PDO $db - the database to insert into.
+ * @param array $catBreeds - the list of cat breeds. As breed id => breed name.
+ * @return void
+ */
+function fillCatBreedToDB(PDO $db, array $catBreeds)
+{
+    foreach($catBreeds as $id => $breed) {
+        $sql = $db->prepare('INSERT INTO `breed` (breed) VALUES (\'' . $breed . '\');');
+        $sql->execute();
+    }
+    echo "Cat breeds added to database.";
+}
+
+/**
+ * Use the list of cat breeds to make the API requests so we can get URLs
+ * of cat pictures to hotlink in the app.
+ * 
+ * @param array of cat breeds
+ * @return array of breed names => array of strings representing the image URLs
+ */
+function getCatImgURLs(array $catBreeds):array
+{
+    $imgSrcArray = [];
+    foreach ($catBreeds as $id => $name) {
+        $catImgApiUrl = 'https://api.thecatapi.com/v1/images/search?breed_ids=' . $id . '&limit=21';
+        $curl = curl_init();
+        curl_setopt_array($curl, array(
+            CURLOPT_URL => $catImgApiUrl,
+            CURLOPT_CUSTOMREQUEST => "GET",
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_HTTPHEADER => array(
+                'x-api-key: ec254e44-3996-458b-8522-4933954d8fcd'
+            ),
+        ));
+        $imgApiResponse = curl_exec($curl);
+        $error = curl_error($curl);
+        curl_close($curl);
+
+        $responseArray = json_decode($imgApiResponse, true);
+
+        $breedImgs = [];
+        foreach($responseArray as $item) {
+            array_push($breedImgs, $item["url"]);
+        }
+        $imgSrcArray[$name] = $breedImgs;
+    }
+    return $imgSrcArray;
+}
+
+/**
+ * Puts the cat image URLs into the database.
+ * 
+ * @param PDO $db the database to use
+ * @param array The list of cat breeds
+ * @param array The list of cat img URLs
+ * @return void
+ */
+function fillCatImgs(PDO $db, array $catBreedArray, array $catImgSrcArray)
+{
+    $catBreedIndexedArray = [];
+    foreach($catBreedArray as $breed) {
+        $catBreedIndexedArray[] = $breed;
+    }
+    $sqlArray = [];
+    for($breedIndex = 0; $breedIndex < count($catImgSrcArray); $breedIndex++) {
+        foreach($catImgSrcArray[$catBreedIndexedArray[$breedIndex]] as $url) {
+            $breedID = $breedIndex + 1;
+            $sqlArray[] = "('$url', $breedID)";
+        }
+    }
+    $sqlString = implode(',', $sqlArray);
+    $sql = $db->prepare('INSERT INTO `img` (img_src, breed_id) VALUES ' . $sqlString . ';');
+    $sql->execute();
+    echo "Cat image URLs added to database.";
+}
+
+// Do the business:
+createDatabase($dbconnection);
+$breeds = getCatBreeds();
+fillCatBreedToDB($dbconnection, $breeds);
+$catImgSrcArray = getCatImgURLs($breeds);
+fillCatImgs($dbconnection, $breeds, $catImgSrcArray);
```

Comment:
All of this stuff is being run procedurally at the bottom of a functional file, and the file itself is being run manually, as a one-off, independently of your app functionality. Where's the value in making this functional? What's being reused?